### PR TITLE
fix: sanitize `<TradeAmountInput />` precision

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR ensures:

- crypto inputs properly honor their asset precision as a maximum allowed dp
- percent option change doesn't result in input amounts with dp greater than their asset's precision
- Tested in:
  - [x] RFOX
  - [x] DeFi PairDeposit (e.g UNI-V2)
  - [x] DeFi Deposit (e.g Cosmos, THORChain savers)
  - [x] DeFi Withdraw (e.g ETH/FOX)
  - [x] Savers
  - [x] Send details
  - [x] Lending
  - [x] THORChain LP

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- closes https://github.com/shapeshift/web/issues/7125

- Low, mostly related to fiat amounts and send modal without initial asset selection

> What protocols, transaction types or contract interactions might be affected by this PR?

### Send Modal

- Amount input is still happy on dashboard send click (no need to broadcast the Tx, only verify that amount input is still happy)
- QR code scanning which prompts for asset selection (e.g ETH receive) is still happy - same thing here, no need to test until broadcast, only ensuring amount input is happy

### Amount Withdraws

Test in:
	- RFOX
	- Savers
	- THOR Lending
	- THOR LP
	- DeFi PairDeposit (e.g UNI-V2)
	- DeFi Deposit (e.g Cosmos, THORChain savers)
	- DeFi Withdraw
	- Send details amount input

For each of these, ensure that:

- Percent option click doesn't result in an amount with more decimal places than the asset's precision (e.g 8 for BTC, 6 for USDC and ATOM, 18 for ETH)
- Amount inputs do not allow entering an amount with more decimal places than the asset's precision
- Toggling fiat does not produce any input regression

### Engineering

- ^

### Operations

- ^

## Screenshots (if applicable)

Tested against develop (right)

- Amount input is still happy on dashboard send click 

https://jam.dev/c/aca1f59b-da64-4278-857d-aa80595ee17c

- QR code scanning which prompts for asset selection (e.g ETH receive) is still happy

image

- RFOX

Screenshot 2024-06-13 at 13 20 53
Screenshot 2024-06-13 at 14 45 55

- DeFi PairDeposit (UNI-V2)

Screenshot 2024-06-13 at 14 51 50
Screenshot 2024-06-13 at 14 54 10

- DeFi Withdraw (ETH/FOX)

Screenshot 2024-06-13 at 14 54 10
Screenshot 2024-06-13 at 14 54 20

- DeFi Deposit (THORChain savers)

Screenshot 2024-06-13 at 14 58 49
Screenshot 2024-06-13 at 14 59 06

- THOR Lending

Screenshot 2024-06-13 at 15 09 52

- THOR LP

Screenshot 2024-06-13 at 15 11 18

- Send details

Screenshot 2024-06-13 at 15 06 54

### CODE REVIEW

File: src/components/DeFi/components/Allocation.tsx
Comment (gomesalexandre) [inline src/components/DeFi/components/Allocation.tsx:1]: Unused

Comment (gomesalexandre) [inline src/components/DeFi/components/Allocation.tsx:1]: Dead code

File: src/features/defi/components/Deposit/PairDepositWithAllocation.tsx
Comment (gomesalexandre) [inline src/features/defi/components/Deposit/PairDepositWithAllocation.tsx:1]: Dead code

**Comment by NeOMakinG** [comment]
## Testing
- [x] RFOX
- [x] Savers
- [x] THOR Lending
- [x] THOR LP
- [x] DeFi PairDeposit (e.g UNI-V2)
- [x] DeFi Deposit (e.g Cosmos, THORChain savers)
- [x] Send details amount input

### Send
Doesn't allow to enter more than 18 decimals for ETH
image
Doesn't allow to enter more than 6 decimals for USDC
image

### rFOX
image

Probably unrelated to the PR but switching to fiat is chaning the FOX amount which feels weird but can't probably be changed due to fiat precision:
image

Unstake is happy with 18 precisions
image

### Savers
BTC:
image
Happy and fiat switch happy too, 8 precisions

ETHEREUM:
image
18 precisions, fiat switch happy

### Thor lending
BTC:
image
Can't test repay as I don't have any amounts

ETH:
image
Same for repay

### Thor LP
image

Both BTC and RUNE are happy with max 8 precisions

ETH+RUNE
image

Both are happy, ETH accepts 18 precisions while rune is stuck at 8, fiat switch ok

### Defi pair deposit univ2
Deposit
image

Withdraw
image

### Defi deposit
image
Fiat switch happy too with 6 precisions

**Review by NeOMakinG — APPROVED** [review]
LGTM, everything seems sane! ✅

**Pull request merged**